sound/soc/sof/intel/hda-stream.c

```diff
@@ -185,6 +185,17 @@ hda_dsp_stream_get(struct snd_sof_dev *sdev, int direction)
 			direction == SNDRV_PCM_STREAM_PLAYBACK ?
 			"playback" : "capture");
 
+	/*
+	 * Disable DMI Link L1 entry when there is capture stream is opened.
+	 * Workaround to address a known issue with host DMA that results
+	 * in xruns during pause/release in capture scenarios.
+	 */
+	if (IS_ENABLED(SND_SOC_SOF_HDA_DISABLE_DMI_L1))
```

Comment:
can we make this if (!IS_ENABLED(SND_SOC_SOF_HDA_ALWAYS_ENABLE_DMI_L1)) ?

so that if the option is not selected by default we get the workaround. You have to explicit select the option to remove the intelligence and enable DM1_L1 for both playback and capture.